src/data/ua_overrides.js

```diff
@@ -549,6 +549,38 @@ const AVAILABLE_UA_OVERRIDES = [
       },
     },
   },
+  {
+    /*
+     * Bug 1598198 - User Agent extension for Samsung's galaxy.store URLs
+     *
+     * Samsung's galaxy.store shortlinks are supposed to redirect to a Samsung
+     * intent:// URL on Samsung devices, but to an error page on other brands.
+     * As we do not provide device info in our user agent string, this check
+     * fails, and even Samsung users land on an error page if they use Firefox
+     * for Android.
+     * This intervention adds a simple "Samsung" identifier to the User Agent
+     * on only the Galaxy Store URLs if the device happens to be a Samsung.
+     */
+    id: "bug1598198",
+    platform: "android",
+    domain: "galaxy.store",
+    bug: "1598198",
+    config: {
+      matches: ["*://galaxy.store/*"],
+      uaTransformer: originalUA => {
+        if (!browser.systemManufacturer) {
+          return originalUA;
+        }
+
+        let manufacturer = browser.systemManufacturer.getManufacturer();
+        if (manufacturer && manufacturer.toLowerCase() == "samsung") {
```

Comment:
Note for posterity: I asked our Samsung contact if it will always return `Samsung` and they said for all the devices they care about, yes. It's possible some really old Samsung phones on Android returned something different, but we don't need to care about them (per Samsung).

Reply:
Ah, that's great, thanks for the info. Then this should be ready after a single safety-check by our Softvision friends. :) I'll prepare builds!